src/main/java/com/urbanairship/datacube/dbharnesses/HBaseDbHarness.java

```diff
@@ -343,53 +361,57 @@ private void flushBatch(Batch<T> batch) throws IOException, InterruptedException
             }
         }
     }
-    
+
     @Override
     public List<Optional<T>> multiGet(List<Address> addresses) throws IOException {
-        int size = addresses.size();
-        List<Get> gets = Lists.newArrayListWithCapacity(size);
-        List<Optional<T>> resultsOptionals = Lists.newArrayListWithCapacity(size);
-        List<byte[]> rowKeys = Lists.newArrayListWithCapacity(size);
+        final int size = addresses.size();
+        final List<Get> gets = Lists.newArrayListWithCapacity(size);
+        final List<Optional<T>> resultsOptionals = Lists.newArrayListWithCapacity(size);
+        final List<byte[]> rowKeys = Lists.newArrayListWithCapacity(size);
+        final Set<Integer> unknownKeyPositions = Sets.newHashSet();
+
+        for (int i = 0; i < addresses.size(); i++) {
+            Address address = addresses.get(i);
 
-        for (Address address : addresses) {
-            final byte[] rowKey;
             try {
-                rowKey = ArrayUtils.addAll(uniqueCubeName, address.toKey(idService));
+                final Optional<byte[]> maybeKey = address.toReadKey(idService);
```

Comment:
It sounds like we are in agreement in that case.

Do you want to just try/catch the call to `address.toReadKey`, since then it's explicit what might throw the InterruptedException in this context?